src/pcp/gadgets.rs

```diff
@@ -2,42 +2,69 @@
 
 //! A collection of gadgets.
 
-use crate::fft::{discrete_fourier_transform, discrete_fourier_transform_inv};
+use crate::fft::{discrete_fourier_transform, discrete_fourier_transform_inv_finish};
 use crate::field::FieldElement;
 use crate::pcp::{Gadget, GadgetCallOnly, PcpError};
+use crate::polynomial::{poly_deg, poly_eval, poly_mul};
+
+use std::convert::TryFrom;
+use std::marker::PhantomData;
+
+const FFT_THRESHOLD_MUL: usize = 60;
```

Comment:
I saw your comment in `speed_tests.rs` explaining how these values were derived. I think we should have a similar comment here.